src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -720,20 +723,30 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
 
     //Determine pet type
     PetType petType = MAX_PET_TYPE;
-    if (IsPet() && owner->GetTypeId() == TYPEID_PLAYER)
-    {
-        if (owner->getClass() == CLASS_WARLOCK ||
-            owner->getClass() == CLASS_SHAMAN ||          // Fire Elemental
-            owner->getClass() == CLASS_DEATH_KNIGHT ||    // Risen Ghoul
-            owner->getClass() == CLASS_MAGE)              // Water Elemental with glyph
-            petType = SUMMON_PET;
-        else if (owner->getClass() == CLASS_HUNTER)
+    if (owner->GetTypeId() == TYPEID_PLAYER)
+    {
+        sScriptMgr->OnBeforeGuardianInitStatsForLevel(owner->ToPlayer(), this, cinfo, petType);
+
+        if (IsPet())
         {
-            petType = HUNTER_PET;
-            m_unitTypeMask |= UNIT_MASK_HUNTER_PET;
+            if (petType == MAX_PET_TYPE)
```

Comment:
can you explain me it?

Reply:
petType is initialized with MAX_PET_TYPE above. If it is not overwritten by the hook it is still MAX_PET_TYPE (the normal case without modules using the hook). If it is MAX_PAT_TYPE it has to continue with normal initialization just as it was before.

Reply:
Cool, maybe you can add a bit of comment for the developer experience. I would love it

Reply:
Ok, pushed a new commit adding an appropriate comment. Hope this suffices.